api/src/main/kotlin/com/google/devtools/ksp/visitor/KSValidateVisitor.kt

```diff
@@ -40,6 +54,9 @@ class KSValidateVisitor(private val predicate: (KSNode, KSNode) -> Boolean) : KS
     }
```

Comment:
Should have noticed previously: Since this visitor extends `KSDefaultVisitor`, why not delegating to it? I.e., replace calls to `validateAnnotations()`, `validateTypeParameters()`, `validateDeclarations()` and `validateDeclaration()` with a single `return super.validateClassDeclaration()`. Then you don't need `private fun validateBlabla`; Just put them in `visitBlabla()`. For example,

```
override fun visitClassDeclaration(c: KSClassDeclaration) {
  // Only validate super types here
  if ( c.superTypes.any { it.accept(this, c) } == false )
    return false
  // This in turn calls visitDeclaration and visitDeclarationContainer
  return super.visitClassDeclaration()
}

override fun visitDeclaration(decl: KSDeclaration) {
  if (decl.typeParameters.any { it.accept(this, decl) } == false)
    return false
  return super.visitDeclaration()
}

override fun visitTypeParameter() {
  // validate a type parameter.
}

override fun visitDeclarationContainer(container: KSDeclarationContainer, ...) {
  return container.declarations.all { it.accept(this, container) }
}

```

and so on.